synapse/storage/_base.py

```diff
@@ -191,6 +192,14 @@ def __init__(self, db_conn, hs):
 
         self.database_engine = hs.database_engine
 
+        if isinstance(self.database_engine, PostgresEngine):
+            self._cache_id_gen = StreamIdGenerator(
+                db_conn, "cache_invalidation_stream", "stream_id",
+            )
+            self.clock = self.hs.get_clock()
+        else:
+            self._cache_id_gen = None
```

Comment:
So, fun fact: this now gets set on the slave replication stores as well as the store on master. This is a bit awkward because we do set `self._cache_id_gen` on the replication stores in `synapse/replication/slave/storage/_base.py` to something different.

In short, everything is awful and is scary and I'm scared about moving this here in case the replication stuff fail to properly overwrite this definition